api/services/stats.py

```diff
@@ -69,8 +70,11 @@ def get_question_stats(user):
 
 
 def get_challenge_stats(user):
+    events = Event.objects.filter(type="CHALLENGE")
+    challenges_solved = sum(event.has_solved_event(user) for event in events)
```

Comment:
The logic here seems fine but as a sanity check, what is `challenges_solved` supposed to mean?

`Event.has_solved_event()` only checks if any of the questions within that event have been solved by the user - the variable name `challenges_solved` implied to me that it means the number of challenges they have solved every question from, but the value you're calculating doesn't seem to be that

Reply:
To clarify here, I'm basically saying I think there is a logic error in what `Event.has_solved_event()` does as a function